libs/designsystem/src/lib/components/page/page.component.spec.ts

```diff
@@ -118,45 +117,46 @@ describe('PageComponent', () => {
   }));
 
   it('should show tab bar when tabBarBottomHidden is false', fakeAsync(() => {
-    // show tab bar
+    // hide tab bar
     spectator.setInput('tabBarBottomHidden', true);
     spectator.detectChanges();
     tick();
     expect(tabBar.tabBarBottomHidden).toBe(true);
 
-    // hide tab bar
+    // show tab bar
     spectator.setInput('tabBarBottomHidden', false);
     spectator.detectChanges();
     tick();
 
     expect(tabBar.tabBarBottomHidden).toBe(false);
   }));
 
-  it('should show tab bar when tabBarBottomHidden on leave', fakeAsync(() => {
+  it('should show tab bar when tabBarBottomHidden is true on leave', fakeAsync(() => {
     spectator.setInput('tabBarBottomHidden', true);
-    spectator.detectChanges();
+    tick();
+    expect(tabBar.tabBarBottomHidden).toBe(true);
 
     triggerOnLeave(router);
 
     expect(tabBar.tabBarBottomHidden).toBe(false);
-    tick();
+    flush();
   }));
 
   describe('onEnter', () => {
-    it('should trigger onEnter when navigating to url', () => {
+    it('should trigger onEnter when navigating to url', async () => {
       const enterEmitSpy = spyOn(spectator.component.enter, 'emit');
 
-      router.navigate(['']);
+      await zone.run(() => router.navigate(['']));
       spectator.detectChanges();
```

Comment:
The order should be reversed: `spectator.detectChanges()` ensures `ngAfterViewInit` is run and nav listeners is set up
```suggestion
      spectator.detectChanges(); // run ngAfterViewInit to setup nav listener
      await zone.run(() => router.navigate(['']));
```